front/src/components/QrReader/components/QrReaderMultiBoxContainer.tsx

```diff
@@ -26,10 +25,11 @@ import {
   FailedBoxesFromMoveBoxesAlert,
   NotInStockAlertText,
 } from "./AlertTexts";
+import { useLoadAndSetGlobalPreferences } from "hooks/useLoadAndSetGlobalPreferences";
 
 function QrReaderMultiBoxContainer() {
-  const { globalPreferences } = useContext(GlobalPreferencesContext);
-  const currentBaseId = globalPreferences.selectedBase?.id;
+  const { baseId } = useLoadAndSetGlobalPreferences();
```

Comment:
put baseId retrieval into a separate helper function or hook to make the code more readible.